pandas/tests/reshape/concat/test_concat.py

```diff
@@ -502,6 +503,26 @@ def test_concat_duplicate_indices_raise(self):
         with pytest.raises(InvalidIndexError, match=msg):
             concat([df1, df2], axis=1)
 
+    def test_concat_with_categorical_indices(self):
+        # GH 44099
+        # concat frames with categorical indices that have the same values
+
+        df1 = DataFrame(
+            {"col1": ["a_val", "b_val", "c_val"]},
+            index=CategoricalIndex(["a", "b", "c"], categories=["a", "b", "c"]),
+        )
+        df2 = DataFrame(
+            {"col1": ["b_val", "a_val", "c_val"]},
+            index=CategoricalIndex(["b", "a", "c"], categories=["b", "a", "c"]),
+        )
+        expected = DataFrame(
+            {"col1": ["a_val", "b_val", "c_val", "b_val", "a_val", "c_val"]},
+            index=CategoricalIndex(
+                ["a", "b", "c", "b", "a", "c"], categories=["a", "b", "c"]
+            ),
+        )
+        tm.assert_frame_equal(concat([df1, df2]), expected)
```

Comment:
use 
```result = concat([df1, df2])```

Reply:
Thanks! I changed at this commit https://github.com/pandas-dev/pandas/pull/46019/commits/91d215568867985d1014f680d60d41f61f33af1e